x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/operator/OperatorPrivileges.java

```diff
@@ -49,6 +54,29 @@ ElasticsearchSecurityException check(
             ThreadContext threadContext
         );
 
+        /**
```

Comment:
following the existing design, `OperatorPrivilegesService` checks to see if the user was flagged as operator during authentication. If the user is an operator, they have no additional restrictions so `OperatorPrivilegesService` will return fast, however if the user is not an operator (i.e. they might be restricted) then delegate down to `OperatorOnlyRegistry` to see if the resource attempting be access is govenerd by operator privs...if so the restrict them.  It's possible to do as you state, but it would end up with differing strategies to enforce operator privs. There is a bit of code duplication, but conceptually all (transport, settings, and now rest) enforcement of operator privs work the same way.